protocol/nats_jetstream/v3/options.go

```diff
@@ -0,0 +1,123 @@
+/*
+ Copyright 2024 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package nats_jetstream
+
+import (
+	"errors"
+
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nats.go/jetstream"
+)
+
+var (
+	ErrNoConnection              = errors.New("URL or nats connection must be given")
+	ErrNoFilterSubjects          = errors.New("no filter subjects were given")
+	ErrMoreThanOneStream         = errors.New("more than one stream for given filter subjects")
+	ErrNoConsumerConfig          = errors.New("no consumer config was given")
+	ErrMoreThanOneConsumerConfig = errors.New("more than one consumer config given")
+)
+
+// ProtocolOption is the function signature required to be considered an nats.ProtocolOption.
+type ProtocolOption func(*Protocol) error
+
+// WithURL configures the Sender and/or Receiver
+func WithURL(url string, natsOpts ...nats.Option) ProtocolOption {
+	return func(p *Protocol) error {
+		opts := []nats.Option{}
+		opts = append(opts, natsOpts...)
+		p.url = url
+		p.natsOpts = opts
+		p.connOwned = true
```

Comment:
Furthermore, I think we can get away without the connOwned field. A protocol user should always call Close on the protocol. Therefore he/she has control over when the connection is closed. I suggest removing this logic here and implementing closing logic in the Close method on the protocol (and document what it does).